File: LOO 2°Parcial/Tarea 1/Person/Person/Teacher.cpp
#include "Teacher.h"
Teacher::Teacher(string nombre, string apellido, int edad, int day, int month, int year, string grade, string type, string courseName, string nrc, int cr) : Persona(nombre, apellido, edad, day, month, year) {
	this->grade = grade;
	this->type = type;
	this->course = new Course(courseName, nrc, cr);
}
void Teacher::setGrade(string grade) {
	this->grade = grade;
}
string Teacher::getGrade(void) {
	return this->grade;
}
void Teacher::setType(string type) {
	this->type = type;
}
string Teacher::getType(void) {
	return this->type;
}

void Teacher::imprimir(void) {
	cout << Persona::toString();
	cout << "\nGRADO ACADEMICO: " << this->grade << "\nTIPO DE DOCENTE: " << this->type;
	this->course->imprimir();
}

File: LOO 2°Parcial/Tarea 1/PersonJava/src/personjava/Teacher.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package personjava;
import  personjava.Course;

/**
 * 
 * @author Dimitri Medina Novelo
 */
public class Teacher extends Persona {
        private String grade;
        private String type;
        private Course curso;
        //private String course;
        
        public Teacher(String nombre, String apellido, int edad, int day, int month, int year,String grade, String type, Course curso){
                super( nombre,  apellido,  edad,  day,  month,  year);
                this.setGrade(grade);
                this.setType(type);
                this.curso = curso;
                //this.setCourse(course);
        }
        public String getGrade() { return this.grade; }
        public void setGrade(String grade) { this.grade = grade; }
        
        public String getType() { return this.type; }
        public void setType(String type) { this.type = type; }
        
        public Course getCurso() {return this.curso;}
       /* public String getCourse() { return this.course; }
        public void setCourse(String course) { this.course = course; }*/
        
        //@Override
        public void imprimir() {
        System.out.println(toString());
        System.out.print("GRADO ACADEMICO: " + this.grade + "\nTIPO DE DOCENTE: " + this.type);
        this.curso.imprimir();
    }
}
